docs-Changes to the CLI `surreal start` command documentation (#818)
surrealdb/docs.surrealdb.com: doc-surrealdb_versioned_docs/version-latest/cli/start.mdx

## Example usage

This example will show how to host a SurrealDB server with the `surreal start` command, and then access the Surreal DB server using the [`surreal sql` command](/docs/surrealdb/cli/sql).

To start a SurrealDB server, run the `surreal start` command, using the options below. This example stores the database in memory, with a username and password, hosted at 127.0.0.1:8000 (the default location).

```bash
surreal start memory --user my_username --pass my_password
```

The following should appear in your console window, with potential slight differences depending on your operating system and version of SurrealDB. The server is actively running, and should be left alone until you want to stop hosting the SurrealDB server.

> Note the message "Started web server on 127.0.0.1:8000", which indicates where the server is being hosted and must be accessed by clients. The location 127.0.0.1:8000 is the default, and can be manually changed by specifing the `--bind` option of the `surreal start` command.

Reviewer: ```suggestion
Note the message "Started web server on 127.0.0.1:8000", which indicates where the server is being hosted and must be accessed by clients. The location `127.0.0.1:8000` is the default, and can be manually changed by specifing the `--bind` option of the `surreal start` command.
```

> ```text
>  .d8888b.                                             888 8888888b.  888888b.
> d88P  Y88b                                            888 888  'Y88b 888  '88b
> Y88b.                                                 888 888    888 888  .88P
>  'Y888b.   888  888 888d888 888d888  .d88b.   8888b.  888 888    888 8888888K.
>     'Y88b. 888  888 888P'   888P'   d8P  Y8b     '88b 888 888    888 888  'Y88b
>       '888 888  888 888     888     88888888 .d888888 888 888    888 888    888
> Y88b  d88P Y88b 888 888     888     Y8b.     888  888 888 888  .d88P 888   d88P
>  'Y8888P'   'Y88888 888     888      'Y8888  'Y888888 888 8888888P'  8888888P'
> 
> 
> 2024-09-04T08:54:48.898733Z  INFO surreal::env: Running 2.0.0-beta.1 for windows on x86_64
> 2024-09-04T08:54:48.899231Z  INFO surrealdb::core::kvs::tr: Starting kvs store in memory
> 2024-09-04T08:54:48.899652Z  INFO surrealdb::core::kvs::tr: Started kvs store in memory
> 2024-09-04T08:54:48.900734Z  INFO surrealdb::core::kvs::tr: Credentials were provided, and no root users were found. The root user 'my_username' will be created
> 2024-09-04T08:54:48.928488Z  INFO surrealdb::net: Started web server on 127.0.0.1:8000
> ```

Reviewer: @matthewjnield we've intentionally left this out as the version is displayed and that can be out of date with the next beta for example 
```suggestion
```

```bash
surreal sql --endpoint http://127.0.0.1:8000 --user my_username --pass my_password --namespace my_namespace --database my_database
```

The following should appear in your console window, with potential slight differences depending on your operating system and version of SurrealDB. This is a Surreal SQL REPL in which you can enter SurrealQL statements and queries. See the documentation of the [`surreal sql` command](/docs/surrealdb/cli/sql) for more information.

```text
#
#  Welcome to the SurrealDB SQL shell
#
#  How to use this shell:
#    - Different statements within a query should be separated by a (;) semicolon.
#    - To create a multi-line query, end your lines with a (\) backslash, and press enter.
#    - To exit, send a SIGTERM or press CTRL+C
#
#  Consult https://surrealdb.com/docs/cli/sql for further instructions
#
#  SurrealDB version: 2.0.0-beta.1

> #

Reviewer: https://github.com/surrealdb/docs.surrealdb.com/pull/818/files#r1743608700

> ```text
> #
> #  Welcome to the SurrealDB SQL shell
> #
> #  How to use this shell:
> #    - Different statements within a query should be separated by a (;) semicolon.
> #    - To create a multi-line query, end your lines with a (\) backslash, and press enter.
> #    - To exit, send a SIGTERM or press CTRL+C
> #
> #  Consult https://surrealdb.com/docs/cli/sql for further instructions
> #
> #  SurrealDB version: 2.0.0-beta.1
> #
> ```

Reviewer: ```suggestion

```

## Example usage

This example will show how to host a SurrealDB server with the `surreal start` command, and then access the Surreal DB server using the [`surreal sql` command](/docs/surrealdb/cli/sql).

To start a SurrealDB server, run the `surreal start` command, using the options below. This example stores the database in memory, with a username and password, hosted at `127.0.0.1:8000` (the default location).

```bash
surreal start memory --user my_username --pass my_password
```

> The following should appear in your console window, with potential slight differences depending on your operating system and version of SurrealDB. The server is actively running, and should be left alone until you want to stop hosting the SurrealDB server.

Reviewer: ```suggestion
The server is actively running, and should be left alone until you want to stop hosting the SurrealDB server.
```

> Note the message "Started web server on 127.0.0.1:8000", which indicates where the server is being hosted and must be accessed by clients. The location `127.0.0.1:8000` is the default, and can be manually changed by specifing the `--bind` option of the `surreal start` command.

Reviewer: ```suggestion
:::note
__Note:__ The message "Started web server on 127.0.0.1:8000", indicates where the server is being hosted and must be accessed by clients. The location `127.0.0.1:8000` is the default, and can be manually changed by specifying the `--bind` option of the `surreal start` command.
:::
```

> To access the SurrealDB server that you have started hosting, open a separate terminal which will act as the client, while the previous terminal is still running the `surreal start` command described above. In the new terminal, run the [`surreal sql` command](/docs/surrealdb/cli/sql) using the options shown below.

Reviewer: ```suggestion
To access the SurrealDB server that you have started hosting, open a separate terminal which will act as the "client", while the previous terminal is still running the `surreal start` command described above. 

In the new terminal, run the [`surreal sql` command](/docs/surrealdb/cli/sql) using the options shown below.
```

Make sure that the hosting location indicated by the output of the `surreal start` command is passed to the `--endpoint` argument, and that you specify the same username and password as in the `surreal start` command. This example also selects a namespace and database so that you can immediately start entering queries if you wish.

```bash
surreal sql --endpoint http://127.0.0.1:8000 --user my_username --pass my_password --namespace my_namespace --database my_database
```

> The following should appear in your console window, with potential slight differences depending on your operating system and version of SurrealDB. This is a Surreal SQL REPL in which you can enter SurrealQL statements and queries. See the documentation of the [`surreal sql` command](/docs/surrealdb/cli/sql) for more information.

Reviewer: ```suggestion
See the documentation of the [`surreal sql` command](/docs/surrealdb/cli/sql) for more information.
```

```bash
surreal start memory --user my_username --pass my_password
```

The server is actively running, and should be left alone until you want to stop hosting the SurrealDB server.

:::note
__Note:__ The message "Started web server on 127.0.0.1:8000", indicates where the server is being hosted and must be accessed by clients. The location `127.0.0.1:8000` is the default, and can be manually changed by specifying the `--bind` option of the `surreal start` command.
:::

To access the SurrealDB server that you have started hosting, open a separate terminal which will act as the "client", while the previous terminal is still running the `surreal start` command described above. 

In the new terminal, run the [`surreal sql` command](/docs/surrealdb/cli/sql) using the options shown below.

Ensure that the hosting location indicated by the output of the `surreal start` command is passed to the `--endpoint` argument, and that you specify the same username and password as in the `surreal start` command.

This example also selects a namespace and database so that you can immediately start entering queries if you wish.

```bash

> surreal sql --endpoint http://127.0.0.1:8000 --namespace my_namespace --database my_database

Reviewer: ```suggestion
surreal sql --endpoint http://127.0.0.1:8000 --namespace my_namespace --database my_database --auth-level root --username root --password root
```

> surreal sql --endpoint http://127.0.0.1:8000 --user my_username --pass my_password --namespace my_namespace --database my_database

Reviewer: ```suggestion
surreal sql --endpoint http://127.0.0.1:8000 --namespace my_namespace --database my_database
```

```

> See the documentation of the [`surreal sql` command](/docs/surrealdb/cli/sql) for more information.
> 
> 

Reviewer: ```suggestion
See the documentation of the [`surreal sql` command](/docs/surrealdb/cli/sql) for more information.
```
